app/recordtransfer/views/profile.py

```diff
@@ -13,96 +13,140 @@
 from django.views.generic import CreateView, UpdateView
 from django_htmx.http import trigger_client_event
 
-from recordtransfer.constants import HtmlIds, QueryParameters
-from recordtransfer.emails import send_user_account_updated
+from recordtransfer.constants import HtmlIds, OtherValues, QueryParameters
 from recordtransfer.forms import UserProfileForm
 from recordtransfer.forms.submission_group_form import SubmissionGroupForm
+from recordtransfer.forms.user_forms import ProfileContactInfoForm
 from recordtransfer.models import InProgressSubmission, Submission, SubmissionGroup, User
 
+LOGGER = logging.getLogger(__name__)
+
+
+class UserProfile(View):
+    """Main profile page - handles GET requests only."""
+
+    def get(self, request: HttpRequest) -> HttpResponse:
+        """Render the user profile page."""
+        account_info_form = UserProfileForm(instance=request.user)
+        contact_info_form = ProfileContactInfoForm(instance=request.user)
+        context = {
+            "account_info_form": account_info_form,
+            "contact_info_form": contact_info_form,
+            "js_context": {
+                # Account Info Form
+                "ID_FIRST_NAME": HtmlIds.ID_FIRST_NAME,
+                "ID_LAST_NAME": HtmlIds.ID_LAST_NAME,
+                "ID_GETS_NOTIFICATION_EMAILS": HtmlIds.ID_GETS_NOTIFICATION_EMAILS,
+                "ID_CURRENT_PASSWORD": HtmlIds.ID_CURRENT_PASSWORD,
+                "ID_NEW_PASSWORD": HtmlIds.ID_NEW_PASSWORD,
+                "ID_CONFIRM_NEW_PASSWORD": HtmlIds.ID_CONFIRM_NEW_PASSWORD,
+                # Contact Info Form
+                "id_province_or_state": HtmlIds.ID_CONTACT_INFO_PROVINCE_OR_STATE,
+                "id_other_province_or_state": HtmlIds.ID_CONTACT_INFO_OTHER_PROVINCE_OR_STATE,
+                "other_province_or_state_value": OtherValues.PROVINCE_OR_STATE,
+                # Submission Group Form
+                "ID_SUBMISSION_GROUP_NAME": HtmlIds.ID_SUBMISSION_GROUP_NAME,
+                # Tables
+                "PAGINATE_QUERY_NAME": QueryParameters.PAGINATE_QUERY_NAME,
+                "ID_IN_PROGRESS_SUBMISSION_TABLE": HtmlIds.ID_IN_PROGRESS_SUBMISSION_TABLE,
+                "IN_PROGRESS_SUBMISSION_TABLE_URL": reverse(
+                    "recordtransfer:in_progress_submission_table"
+                ),
+                "ID_SUBMISSION_GROUP_TABLE": HtmlIds.ID_SUBMISSION_GROUP_TABLE,
+                "SUBMISSION_GROUP_TABLE_URL": reverse("recordtransfer:submission_group_table"),
+            },
+        }
 
-class UserProfile(UpdateView):
-    """View to show two things:
-    - The user's profile information
-    - A list of the Submissions a user has made.
-    """
+        return render(request, "recordtransfer/profile.html", context)
 
-    template_name = "recordtransfer/profile.html"
-    form_class = UserProfileForm
+
+class BaseUserProfileUpdateView(UpdateView):
+    """Base view for updating user profile information."""
+
+    model = User
```

Comment:
Including class members instead of the functions `get_success_message` and `get_error_message` is a more Django way of doing things:

```suggestion
class BaseUserProfileUpdateView(UpdateView):
    """Base view for updating user profile information."""

    model = User
    update_success_message = gettext("Updated successfully")
    update_error_message = gettext("Could not update")
```